ui/app/components/clients/page/counts.ts

```diff
@@ -0,0 +1,166 @@
+          title: 'No billing start date found',
+          message:
+            'In order to get the most from this data, please enter your billing period start month. This will ensure that the resulting data is accurate.',
+        }
+      : {
+          label: 'Client counting start date',
+          description:
+            'This date is when client counting starts. Without this starting point, the data shown is not reliable.',
+          title: 'No start date found',
+          message:
+            'In order to get the most from this data, please enter a start month above. Vault will calculate new clients starting from that month.',
+        };
+  }
+
+  get namespaces() {
+    return this.args.activity.byNamespace
+      ? this.args.activity.byNamespace.map((namespace) => ({
+          name: namespace.label,
+          id: namespace.label,
+        }))
+      : [];
+  }
+
+  get authMounts() {
+    if (this.namespaces.length) {
+      return this.activityForNamespace?.mounts.map((mount) => ({
+        id: mount.label,
+        name: mount.label,
+      }));
+    }
+    return [];
+  }
+
+  get startTimeDiscrepancy() {
+    // show banner if startTime returned from activity log (response) is after the queried startTime
+    const { activity, config } = this.args;
+    const activityStartDateObject = parseAPITimestamp(activity.startTime) as Date;
+    const queryStartDateObject = parseAPITimestamp(this.startDate) as Date;
+    const isEnterprise =
+      this.startDate === config.billingStartTimestamp.toISOString() && this.version.isEnterprise;
+    const message = isEnterprise ? 'Your license start date is' : 'You requested data from';
+
+    if (
+      isAfter(activityStartDateObject, queryStartDateObject) &&
+      !isSameMonth(activityStartDateObject, queryStartDateObject)
+    ) {
+      return `${message} ${this.formattedStartDate}.
+        We only have data from ${parseAPITimestamp(activity.startTime, 'MMMM yyyy')},
+        and that is what is being shown here.`;
+    } else {
+      return null;
+    }
+  }
+
+  get activityForNamespace() {
+    const { activity, namespace } = this.args;
+    return namespace ? activity.byNamespace.find((ns) => ns.label === namespace) : null;
+  }
+
+  get filteredActivity() {
+    // return activity counts based on selected namespace and auth mount values
+    const { namespace, authMount, activity } = this.args;
+    if (namespace) {
+      return authMount
+        ? this.activityForNamespace?.mounts.find((mount) => mount.label === authMount)
+        : this.activityForNamespace;
+    }
+    return activity.total;
+  }
+
+  @action
+  onDateChange(dateObject: { dateType: string; monthIdx: string; year: string }) {
+    const { dateType, monthIdx, year } = dateObject;
+    const { currentTimestamp } = this.args;
+    const selectedDate = formatDateObject({ monthIdx, year }, dateType === 'endDate');
```

Comment:
Converts the selections to unix time for the query params. The activity endpoint supports passing the timestamps in this format.

Reply:
Could we add a comment that formatDateObject converts to unix? The date formatting has changed so much over with each iteration, it's helpful to know here that this is the format when it's selected

Reply:
Yep for sure!